### ISSUE TITLE
Starting the process to developing a Webmail client for Maildir systems

### ISSUE THREAD

**Issue by philiprhoades** [WebAssembly/WASI#548]
People,

I didn't get a response to my previous question but I received a reply from an email to a developer and he suggested starting with C (which I have some old experience with) or Rust (which I have no experience with), converting the C program to WASM and then converting that WASM program back to WAT for studying to improve my understanding of WAT - which sounds like a good idea.  However, my longer-term aim is a WASM Webmail client - so my question is now:

Is it going to be possible to convert a WASM program that makes use of WASI back to WAT?  My preference ATM is to just learn to program in WAT . . 

Thanks,

Phil.

**Comment by sbc100** [comment]
wat is not really a practical language in which to author real programs.   Think of it more as an assembly language.

Its easy to convert any wasm file to wat using tools such as wat2wasm (part of wabt) or wasm-dis (part of binaryen).

**Comment by abrown** [comment]
I completely agree with @sbc100 that you probably want to use a higher-level language (but... if you're looking for a challenge, it is possible to handcraft WAT youself: https://github.com/binji/raw-wasm).

**Comment by philiprhoades** [comment]
> Think of it more as an assembly language.

But I always wanted to learn ASM x86 but never got around to it!

> Its easy to convert any wasm file to wat using tools such as wat2wasm (part of wabt) or wasm-dis (part of binaryen).

Yep, doing that with WASMs from little C programs now to see what they look like . . 

> if you're looking for a challenge, it is possible to handcraft WAT youself:

Thanks!

**Comment by sbc100** [comment]
Can we close this issue?

**Comment by philiprhoades** [comment]
> Can we close this issue?

Sure.
